colossalai/booster/booster.py

```diff
@@ -55,50 +55,71 @@ def __init__(self,
                  device: str = 'cuda',
                  mixed_precision: Union[MixedPrecision, str] = None,
                  plugin: Optional[Plugin] = None) -> None:
-        # TODO(FrankLeeeee): add plugin control logic
-        # if self.plugin is not None and self.plugin.control_accelerator:
-        #     ...
-        # create acclerator
-        self.acceleartor = Accelerator(device)
-        self.acceleartor.set_default_device()
-
-        # validate and set precision
-        if isinstance(MixedPrecision, str):
-            # the user will take the default arguments for amp training
-            self.mixed_precision = mixed_precision_factory(mixed_precision)
-        elif isinstance(mixed_precision, MixedPrecision):
-            # the user can customize the arguments by passing the precision object
-            self.mixed_precision = mixed_precision
+        self.plugin = plugin
```

Comment:
When can the plugin be `None`?

Reply:
I have added the type check to ensure `plugin` can only be a `Plugin` object or None. The plugin can be `None` if the user just does not define any acceleration plugin and solely wants to use `mixed_precision`.